release.md

```diff
@@ -0,0 +1,93 @@
+# Release procedure
+
+## Branching
+
+1. Checkout `master` branch.
+   
+    `git checkout master`
+   
+1. Make sure you have the latest upstream changes.
+
+    `git pull upstream master`
+
+1. Create a new release branch.
+
+    `git checkout -b v1.x`
+   
+1. Push branch to the upstream.
+
+    `git push upstream v1.x`
+
+1. Set branch protection rules on `v1.x` branch in GitHub repository settings.
+1. Update release schedule on `master` branch in file`docs/source/release.md`.
+1. Create a PR targeting master branch.
+1. Once PR with release schedule and enabled docs is merged to master, tag merge commit as `alpha.0` of next expected release.
+   Do not merge any other PR to master before the PR merges and `alpha.0` tag is created.
+   
+   `git tag v1.y.0-alpha.0 -m "Alpha release v1.y.0-alpha.0`
+   
+1. On release branch change default value of image tag variable in Makefile. Set only major and minor part of version without the 'v' prefix.
+
+    `IMAGE_TAG ?=1.x`
+   
+1. Commit the change
+
+   `git commit -m "Pin tags to v1.x-rc.0`
+   
+1. Update manifests, examples and chart defaults.
+
+    `make update`
+
+1. Commit changes and create a PR with `v1.x` as base branch.
+   
+   `git commit -a -m "Updated autogenerated code"`
+   
+1. Once PR is merged, tag merge commit as `beta.0` using annotated tag.
+   
+   `git tag v1.x.0-beta.0 -m "Beta release v1.x.0-beta.0`
+   
+1. Enable building docs from `v1.x` branch by adding entry to `docs/source/conf.py` (in the master branch).
+
+    `BRANCHES = ['master', 'v0.3', 'v1.0', 'v1.1', 'v1.x']`
+   
+1. Submit PR with additional docs branch with `master` as target branch.
+
+## Tagging a release candidate
+
+1. Release candidate should be tagged only when all issues in GitHub milestone are closed, and backported to release branch. 
+1. Tag commit using annotated tag
+
+    `git tag v1.x.0-rc.0 <commit-sha> -m "Release candidate v1.x.0-rc.0`
```

Comment:
I think I'd be ok with just `-m "v1.x.0-rc.0`` but this is fine if you prefer it